ext/lmdb-safe/lmdb-typed.cc

```diff
@@ -13,18 +13,20 @@ uint32_t MDBGetMaxID(MDBRWTransaction& txn, MDBDbi& dbi)
   return maxid;
 }
 
-uint32_t MDBGetRandomID(MDBRWTransaction& txn, MDBDbi& dbi)
+uint32_t MDBGetRandomID(MDBRWTransaction& txn, MDBDbi& dbi, uint32_t seed)
 {
   auto cursor = txn->getRWCursor(dbi);
-  uint32_t newID = 0;
+  uint32_t newID = seed & ~(1 << 31);
```

Comment:
```suggestion
  uint32_t newID = seed & ~(1U << 31);
```
...or you would shift into the sign bit and that makes the C standard unhappy.

Reply:
the resulting assembly does not change, but yes, good idea

Reply:
Actually, nevermind, I keep forgetting that this is only a C problem, and this construct is valid in C++.